src/main/java/sg/ncl/MainController.java

```diff
@@ -1323,6 +1342,14 @@ public String userSideRejectJoinRequest(
                         log.warn("Reject join request: User {}, Team {} fail", userId, teamId);
                         redirectAttributes.addFlashAttribute(MESSAGE, "Reject join request fail");
                         break;
+                    case OPENSTACK_CONNECTION_EXCEPTION:
+                        log.warn("Reject join request: User {}, Team {} fail on OpenStack connection", userId, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE, "Reject join request fail");
+                        break;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Reject join request fail" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)